e2e/helpers.py

```diff
@@ -16,22 +16,32 @@ def pykube_config():
 class KubeHelper:
     def __init__(self):
         self._api = pykube.HTTPClient(pykube_config())
+        self._namespaces = set()
         self._ns = f"secret-sync-e2etest-{uuid.uuid4()}"
 
-    def _get_namespace(self):
-        return pykube.Namespace.objects(self._api).get(name=self._ns)
+    def _get_namespace(self, name):
+        return pykube.Namespace.objects(self._api).get(name=name)
 
-    def _create_namespace(self):
-        body = {"metadata": {"name": self._ns}}
+    def _create_namespace(self, name):
+        self._namespaces.add(name)
+        body = {"metadata": {"name": name}}
         return pykube.Namespace(self._api, body).create()
 
-    def _delete_namespace(self):
-        nsobj = self._get_namespace()
-        if nsobj:
-            return nsobj.delete()
+    def _delete_namespaces(self):
+        for name in self._namespaces:
+            nsobj = self._get_namespace(name)
```

Comment:
will this ever return None, I thought pykube raised an `ObjectNotFound` exception?

Reply:
Good catch, I'll drop the check.

That method's only ever called during cleanup for things that we expect have been created during setup (or the test run), so it's safe to assume the namespaces will already exist. (If they didn't our tests would already be failing.)